roxie/udplib/udpsha.cpp

```diff
@@ -34,18 +34,38 @@ using roxiemem::IDataBufferManager;
 
 IDataBufferManager *bufferManager;
 
+// All exported udp configuration options - these values provide the default values
+#ifdef TEST_DROPPED_PACKETS
+bool udpDropDataPackets = false;
+unsigned udpDropFlowPackets[flowType::max_flow_cmd] = {};
+unsigned flowPacketsSent[flowType::max_flow_cmd] = {};
+#endif
+
 bool udpTraceFlow = false;
 bool udpTraceTimeouts = false;
+
 unsigned udpTraceLevel = 0;
 unsigned udpFlowSocketsSize = 131072;
 unsigned udpLocalWriteSocketSize = 1024000;
 unsigned udpStatsReportInterval = 60000;
 
-#ifdef TEST_DROPPED_PACKETS
-bool udpDropDataPackets = false;
-unsigned udpDropFlowPackets[flowType::max_flow_cmd] = {};
-unsigned flowPacketsSent[flowType::max_flow_cmd] = {};
-#endif
+unsigned udpOutQsPriority = 0;
+unsigned udpMaxRetryTimedoutReqs = 0;   // 0 means off (keep retrying forever)
+
+//The following control the timeouts within the udp layer.  All timings are in milliseconds, but I suspect some of these should possibly be sub-millisecond
+unsigned udpFlowAckTimeout = 2;         // the maximum time that it is expected to take to receive an acknowledgement of a flow message (when one is sent) - should be small
+unsigned updDataSendTimeout = 20;       // how long to receive the maximum amount of data, ~100 packets of 8K should take 10ms on a 1Gb network. Timeout for assuming send_complete has been lost
```

Comment:
Note: The comment suggests that this timeout needs to be dependent on the queue size (or the max slots per client) and possibly the MTU size in use.